src/index.ts

```diff
@@ -347,18 +348,29 @@ export class MayanRoute<N extends Network>
   }
 
   public override async *track(receipt: R, timeout?: number) {
-    if (!isSourceInitiated(receipt)) throw new Error("Transfer not initiated");
-
-    const txstatus = await getTransactionStatus(
-      receipt.originTxs[receipt.originTxs.length - 1]!
-    );
-    if (!txstatus) return;
-
-    receipt = txStatusToReceipt(txstatus);
-
-    // TODO: loop until timeout?
+    // What should be the default if no timeout is provided?
+    let leftover = timeout ? timeout : 60 * 60 * 1000;
+    while (leftover > 0) {
+      const start = Date.now();
+      if (!isSourceInitiated(receipt))
+        throw new Error("Transfer not initiated");
+
+      const txstatus = await getTransactionStatus(
+        receipt.originTxs[receipt.originTxs.length - 1]!
+      );
+      if (!txstatus) continue;
```

Comment:
I think you want to sleep here too before continuing, right? otherwise you will hammer the endpoint while it's 404ing